src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/CohostOnAutoInsertEndpointTest.cs

```diff
@@ -0,0 +1,275 @@
+            This is a Razor document.
+
+            <PageTitle style="$0"></PageTitle>
+
+            The end.
+            """;
+        await VerifyOnAutoInsertAsync(input, output, triggerCharacter: "=", delegatedResponseText: "\"$0\"");
+    }
+
+    [Fact]
+    public async Task Component_AutoInsertCSharp_OnForwardSlash()
+    {
+        var input = """
+        @code {
+            ///$$
+            void TestMethod() {}
+        }
+        """;
+
+        var output = """
+        @code {
+            /// <summary>
+            /// $0
+            /// </summary>
+            void TestMethod() {}
+        }
+        """;
+        await VerifyOnAutoInsertAsync(input, output, triggerCharacter: "/");
+    }
+
+    [Fact]
+    public async Task Component_DoNotAutoInsertCSharp_OnForwardSlashWithFormatOnTypeDisabled()
+    {
+        var input = """
+        @code {
+            ///$$
+            void TestMethod() {}
+        }
+        """;
+
+        var output = """
+        @code {
+            ///
+            void TestMethod() {}
+        }
+        """;
+        await VerifyOnAutoInsertAsync(input, output, triggerCharacter: "/", formatOnType: false, expectResult: false);
+    }
+
+    [Fact]
+    public async Task Component_AutoInsertCSharp_OnEnter()
+    {
+        var input = """
+        @code {
+            void TestMethod() {
+        $$}
+        }
+        """;
+
+        var output = """
+        @code {
+            void TestMethod()
+            {
+                $0
+            }
+        }
+        """;
+        await VerifyOnAutoInsertAsync(input, output, triggerCharacter: "\n");
+    }
+
+    [Fact]
+    public async Task Component_AutoInsertCSharp_OnEnter_TwoSpaceIndent()
+    {
+        var input = """
+        @code {
+            void TestMethod() {
+        $$}
+        }
+        """;
+
+        var output = """
+        @code {
+          void TestMethod()
+          {
+            $0
+          }
+        }
+        """;
+        await VerifyOnAutoInsertAsync(input, output, triggerCharacter: "\n", tabSize: 2);
+    }
+
+    [Fact]
+    public async Task Component_AutoInsertCSharp_OnEnter_UseTabs()
+    {
+        var input = """
+        @code {
+            void TestMethod() {
+        $$}
+        }
+        """;
+
+        var tab = '\t';
+        var output = $$"""
+        @code {
+        {{tab}}void TestMethod()
+        {{tab}}{
+        {{tab}}{{tab}}$0
+        {{tab}}}
+        }
+        """;
+        await VerifyOnAutoInsertAsync(input, output, triggerCharacter: "\n", insertSpaces: false);
+    }
+
+    private async Task VerifyOnAutoInsertAsync(
+        string input,
+        string output,
+        string triggerCharacter,
+        string? delegatedResponseText = null,
+        bool insertSpaces = true,
+        int tabSize = 4,
+        bool formatOnType = true,
+        bool autoClosingTags = true,
+        bool expectResult = true)
+    {
+        var testCode = new TestCode(input);
```

Comment:
Note that `TestCode` is defined so that it is implicitly convertible from `string`. So, you could just declare the `input` parameter as `TestCode` and remove the first line of this method.

```suggestion
        TestCode input,
        string output,
        string triggerCharacter,
        string? delegatedResponseText = null,
        bool insertSpaces = true,
        int tabSize = 4,
        bool formatOnType = true,
        bool autoClosingTags = true,
        bool expectResult = true)
    {
```

Then, just update the three references to `testCode` below to `input`.